app/models/source.rb

```diff
@@ -1,15 +1,17 @@
 class Source < ApplicationRecord
-  validates_lengths_from_database
   has_many :reports, :through => :logs
   has_many :logs
-  validates :value, :digest, :presence => true
+  validates_lengths_from_database
 
   def to_s
     value
   end
 
-  def self.find_or_create(val)
-    digest = Digest::SHA1.hexdigest(val)
-    Source.find_by(:digest => digest) || Source.create(:value => val, :digest => digest)
+  def self.find_or_create(value)
+    Source.find_or_create_by(value: value)
+  end
+
+  def skip_strip_attrs
+    ['value']
   end
```

Comment:
Skip attribute stripping too to match message model, also saves few CPU cycles during processing as there are many of these usually.